lightning/src/chain/keysinterface.rs

```diff
@@ -245,12 +247,55 @@ pub trait ChannelKeys : Send+Clone {
 	/// return value must contain a signature.
 	fn sign_local_commitment_htlc_transactions<T: secp256k1::Signing + secp256k1::Verification>(&self, local_commitment_tx: &LocalCommitmentTransaction, local_csv: u16, secp_ctx: &Secp256k1<T>) -> Result<Vec<Option<Signature>>, ()>;
 
+	/// Create a signature for a justice transaction spending a revokable output on a revoked
+	/// remote commitment or HTLC transaction.
+	///
+	/// Justice transaction may claim multiples outputs at same time if timelock are similar.
+	/// It may be called multiples time for same output(s) if a fee-bump is needed with regards
+	/// to an upcoming timelock expiration.
+	///
+	/// Witness_script is a revokable witness script as defined in BOLT3 for `to_local`/HTLC
+	/// outputs.
+	///
+	/// Amount is value of the output spent by this input.
+	///
+	/// Per_commitment key is revocation secret such as provided by remote party while
+	/// revocating detected onchain transaction. It's not a _local_ secret key, therefore
+	/// it may cross interfaces, a node compromise won't allow to spend revoked output without
+	/// also compromissing revocation key.
+	//TODO: replace revocation_pubkey by per_commitment_point
+	//TODO: move to Result<Signature, ()>
+	fn sign_justice_transaction<T: secp256k1::Signing>(&self, justice_tx: &mut Transaction, input: usize, witness_script: &Script, amount: u64, per_commitment_key: &SecretKey, revocation_pubkey: &PublicKey, is_htlc: bool, secp_ctx: &Secp256k1<T>);
+
+	/// Create a signature for a claiming transaction for a HTLC output on a remote commitment
+	/// transaction, either offered or received.
+	///
+	/// HTLC transaction may claim multiples offered outputs at same time if we know preimage
+	/// for each at detection. It may be called multtiples time for same output(s) if a fee-bump
+	/// is needed with regards to an upcoming timelock expiration.
+	///
+	/// Witness_script is either a offered or received script as defined in BOLT3 for HTLC
```

Comment:
Same here - if we already have it at the callsite, don't pass the script, pass the data and expose chan_utils::get*redeemscript.